mllib/src/main/scala/org/apache/spark/ml/tuning/ValidatorParams.scala

```diff
@@ -67,6 +71,39 @@ private[ml] trait ValidatorParams extends HasSeed with Params {
   /** @group getParam */
   def getEvaluator: Evaluator = $(evaluator)
 
+  /**
+   * param to control the number of models evaluated in parallel
+   * Default: 1
+   *
+   * @group param
+   */
+  val numParallelEval: IntParam = new IntParam(this, "numParallelEval",
+    "max number of models to evaluate in parallel, 1 for serial evaluation",
+    ParamValidators.gtEq(1))
+
+  /** @group getParam */
+  def getNumParallelEval: Int = $(numParallelEval)
+
+  /**
+   * Creates a execution service to be used for validation, defaults to a thread-pool with
+   * size of `numParallelEval`
+   */
+  protected var executorServiceFactory: (Int) => ExecutorService = {
+    (requestedMaxThreads: Int) => ThreadUtils.newDaemonCachedThreadPool(
```

Comment:
I think it may be problematic (and probably unnecessary) to have daemon threads here - we don't have a shutdown hook.

We could add a shutdown hook to tie the lifecycle to SparkContext (SparkSession); perhaps in `setExecutorService`:

```scala
...
ShutdownHookManager.addShutdownHook(() => executorService.shutdown())
...
```